File: csharp-dotnet2/src/main/CsharpDotNet2/estuary-client/Api/MinerApi.cs
using System;
using System.Collections.Generic;
using RestSharp;
using IO.Swagger.Client;
using IO.Swagger.Model;

namespace IO.Swagger.Api
{
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public interface IMinerApi
    {
        /// <summary>
        /// Get Claim Miner Message This endpoint lets a user get the message in order to claim a miner
        /// </summary>
        /// <param name="miner">Miner claim message</param>
        /// <returns>ApiClaimMsgResponse</returns>
        ApiClaimMsgResponse MinerClaimMinerGet (string miner);
        /// <summary>
        /// Claim Miner This endpoint lets a user claim a miner
        /// </summary>
        /// <param name="body">Claim Miner Body</param>
        /// <returns>ApiClaimResponse</returns>
        ApiClaimResponse MinerClaimPost (MinerClaimMinerBody body);
        /// <summary>
        /// Set Miner Info This endpoint lets a user set miner info.
        /// </summary>
        /// <param name="body">Miner set info params</param>
        /// <param name="miner">Miner to set info for</param>
        /// <returns>ApiEmptyResp</returns>
        ApiEmptyResp MinerSetInfoMinerPut (MinerMinerSetInfoParams body, string miner);
        /// <summary>
        /// Suspend Miner This endpoint lets a user suspend a miner.
        /// </summary>
        /// <param name="body">Suspend Miner Body</param>
        /// <param name="miner">Miner to suspend</param>
        /// <returns>ApiEmptyResp</returns>
        ApiEmptyResp MinerSuspendMinerPost (MinerSuspendMinerBody body, string miner);
        /// <summary>
        /// Unuspend Miner This endpoint lets a user unsuspend a miner.
        /// </summary>
        /// <param name="miner">Miner to unsuspend</param>
        /// <returns>ApiEmptyResp</returns>
        ApiEmptyResp MinerUnsuspendMinerPut (string miner);
        /// <summary>
        /// Get all miners deals This endpoint returns all miners deals
        /// </summary>
        /// <param name="miner">Filter by miner</param>
        /// <param name="ignoreFailed">Ignore Failed</param>
        /// <returns>string</returns>
        string PublicMinersDealsMinerGet (string miner, string ignoreFailed);
        /// <summary>
        /// Get miner stats This endpoint returns miner stats
        /// </summary>
        /// <param name="miner">Filter by miner</param>
        /// <returns>string</returns>
        string PublicMinersStatsMinerGet (string miner);
    }
  
    /// <summary>
    /// Represents a collection of functions to interact with the API endpoints
    /// </summary>
    public class MinerApi : IMinerApi
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="MinerApi"/> class.
        /// </summary>
        /// <param name="apiClient"> an instance of ApiClient (optional)</param>
        /// <returns></returns>
        public MinerApi(ApiClient apiClient = null)
        {
            if (apiClient == null) // use the default one in Configuration
                this.ApiClient = Configuration.DefaultApiClient; 
            else
                this.ApiClient = apiClient;
        }
    
        /// <summary>
        /// Initializes a new instance of the <see cref="MinerApi"/> class.
        /// </summary>
        /// <returns></returns>
        public MinerApi(String basePath)
        {
            this.ApiClient = new ApiClient(basePath);
        }
    
        /// <summary>
        /// Sets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public void SetBasePath(String basePath)
        {
            this.ApiClient.BasePath = basePath;
        }
    
        /// <summary>
        /// Gets the base path of the API client.
        /// </summary>
        /// <param name="basePath">The base path</param>
        /// <value>The base path</value>
        public String GetBasePath(String basePath)
        {
            return this.ApiClient.BasePath;
        }
    
        /// <summary>
        /// Gets or sets the API client.
        /// </summary>
        /// <value>An instance of the ApiClient</value>
        public ApiClient ApiClient {get; set;}
    
        /// <summary>
        /// Get Claim Miner Message This endpoint lets a user get the message in order to claim a miner
        /// </summary>
        /// <param name="miner">Miner claim message</param>
        /// <returns>ApiClaimMsgResponse</returns>
        public ApiClaimMsgResponse MinerClaimMinerGet (string miner)
        {
            // verify the required parameter 'miner' is set
            if (miner == null) throw new ApiException(400, "Missing required parameter 'miner' when calling MinerClaimMinerGet");
    
            var path = "/miner/claim/{miner}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "miner" + "}", ApiClient.ParameterToString(miner));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                    
            // authentication setting, if any
            String[] authSettings = new String[] { "bearerAuth" };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerClaimMinerGet: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerClaimMinerGet: " + response.ErrorMessage, response.ErrorMessage);
    
            return (ApiClaimMsgResponse) ApiClient.Deserialize(response.Content, typeof(ApiClaimMsgResponse), response.Headers);
        }
    
        /// <summary>
        /// Claim Miner This endpoint lets a user claim a miner
        /// </summary>
        /// <param name="body">Claim Miner Body</param>
        /// <returns>ApiClaimResponse</returns>
        public ApiClaimResponse MinerClaimPost (MinerClaimMinerBody body)
        {
            // verify the required parameter 'body' is set
            if (body == null) throw new ApiException(400, "Missing required parameter 'body' when calling MinerClaimPost");
    
            var path = "/miner/claim";
            path = path.Replace("{format}", "json");
                
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                    postBody = ApiClient.Serialize(body); // http body (model) parameter

            // authentication setting, if any
            String[] authSettings = new String[] { "bearerAuth" };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.POST, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerClaimPost: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerClaimPost: " + response.ErrorMessage, response.ErrorMessage);
    
            return (ApiClaimResponse) ApiClient.Deserialize(response.Content, typeof(ApiClaimResponse), response.Headers);
        }
    
        /// <summary>
        /// Set Miner Info This endpoint lets a user set miner info.
        /// </summary>
        /// <param name="body">Miner set info params</param>
        /// <param name="miner">Miner to set info for</param>
        /// <returns>ApiEmptyResp</returns>
        public ApiEmptyResp MinerSetInfoMinerPut (MinerMinerSetInfoParams body, string miner)
        {
            // verify the required parameter 'body' is set
            if (body == null) throw new ApiException(400, "Missing required parameter 'body' when calling MinerSetInfoMinerPut");
            // verify the required parameter 'miner' is set
            if (miner == null) throw new ApiException(400, "Missing required parameter 'miner' when calling MinerSetInfoMinerPut");
    
            var path = "/miner/set-info/{miner}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "miner" + "}", ApiClient.ParameterToString(miner));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                    postBody = ApiClient.Serialize(body); // http body (model) parameter

            // authentication setting, if any
            String[] authSettings = new String[] { "bearerAuth" };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.PUT, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerSetInfoMinerPut: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerSetInfoMinerPut: " + response.ErrorMessage, response.ErrorMessage);
    
            return (ApiEmptyResp) ApiClient.Deserialize(response.Content, typeof(ApiEmptyResp), response.Headers);
        }
    
        /// <summary>
        /// Suspend Miner This endpoint lets a user suspend a miner.
        /// </summary>
        /// <param name="body">Suspend Miner Body</param>
        /// <param name="miner">Miner to suspend</param>
        /// <returns>ApiEmptyResp</returns>
        public ApiEmptyResp MinerSuspendMinerPost (MinerSuspendMinerBody body, string miner)
        {
            // verify the required parameter 'body' is set
            if (body == null) throw new ApiException(400, "Missing required parameter 'body' when calling MinerSuspendMinerPost");
            // verify the required parameter 'miner' is set
            if (miner == null) throw new ApiException(400, "Missing required parameter 'miner' when calling MinerSuspendMinerPost");
    
            var path = "/miner/suspend/{miner}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "miner" + "}", ApiClient.ParameterToString(miner));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                    postBody = ApiClient.Serialize(body); // http body (model) parameter

            // authentication setting, if any
            String[] authSettings = new String[] { "bearerAuth" };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.POST, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerSuspendMinerPost: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerSuspendMinerPost: " + response.ErrorMessage, response.ErrorMessage);
    
            return (ApiEmptyResp) ApiClient.Deserialize(response.Content, typeof(ApiEmptyResp), response.Headers);
        }
    
        /// <summary>
        /// Unuspend Miner This endpoint lets a user unsuspend a miner.
        /// </summary>
        /// <param name="miner">Miner to unsuspend</param>
        /// <returns>ApiEmptyResp</returns>
        public ApiEmptyResp MinerUnsuspendMinerPut (string miner)
        {
            // verify the required parameter 'miner' is set
            if (miner == null) throw new ApiException(400, "Missing required parameter 'miner' when calling MinerUnsuspendMinerPut");
    
            var path = "/miner/unsuspend/{miner}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "miner" + "}", ApiClient.ParameterToString(miner));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                    
            // authentication setting, if any
            String[] authSettings = new String[] { "bearerAuth" };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.PUT, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerUnsuspendMinerPut: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling MinerUnsuspendMinerPut: " + response.ErrorMessage, response.ErrorMessage);
    
            return (ApiEmptyResp) ApiClient.Deserialize(response.Content, typeof(ApiEmptyResp), response.Headers);
        }
    
        /// <summary>
        /// Get all miners deals This endpoint returns all miners deals
        /// </summary>
        /// <param name="miner">Filter by miner</param>
        /// <param name="ignoreFailed">Ignore Failed</param>
        /// <returns>string</returns>
        public string PublicMinersDealsMinerGet (string miner, string ignoreFailed)
        {
            // verify the required parameter 'miner' is set
            if (miner == null) throw new ApiException(400, "Missing required parameter 'miner' when calling PublicMinersDealsMinerGet");
    
            var path = "/public/miners/deals/{miner}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "miner" + "}", ApiClient.ParameterToString(miner));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
             if (ignoreFailed != null) queryParams.Add("ignore-failed", ApiClient.ParameterToString(ignoreFailed)); // query parameter
                        
            // authentication setting, if any
            String[] authSettings = new String[] { "bearerAuth" };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling PublicMinersDealsMinerGet: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling PublicMinersDealsMinerGet: " + response.ErrorMessage, response.ErrorMessage);
    
            return (string) ApiClient.Deserialize(response.Content, typeof(string), response.Headers);
        }
    
        /// <summary>
        /// Get miner stats This endpoint returns miner stats
        /// </summary>
        /// <param name="miner">Filter by miner</param>
        /// <returns>string</returns>
        public string PublicMinersStatsMinerGet (string miner)
        {
            // verify the required parameter 'miner' is set
            if (miner == null) throw new ApiException(400, "Missing required parameter 'miner' when calling PublicMinersStatsMinerGet");
    
            var path = "/public/miners/stats/{miner}";
            path = path.Replace("{format}", "json");
            path = path.Replace("{" + "miner" + "}", ApiClient.ParameterToString(miner));
    
            var queryParams = new Dictionary<String, String>();
            var headerParams = new Dictionary<String, String>();
            var formParams = new Dictionary<String, String>();
            var fileParams = new Dictionary<String, FileParameter>();
            String postBody = null;
    
                                    
            // authentication setting, if any
            String[] authSettings = new String[] { "bearerAuth" };
    
            // make the HTTP request
            IRestResponse response = (IRestResponse) ApiClient.CallApi(path, Method.GET, queryParams, postBody, headerParams, formParams, fileParams, authSettings);
    
            if (((int)response.StatusCode) >= 400)
                throw new ApiException ((int)response.StatusCode, "Error calling PublicMinersStatsMinerGet: " + response.Content, response.Content);
            else if (((int)response.StatusCode) == 0)
                throw new ApiException ((int)response.StatusCode, "Error calling PublicMinersStatsMinerGet: " + response.ErrorMessage, response.ErrorMessage);
    
            return (string) ApiClient.Deserialize(response.Content, typeof(string), response.Headers);
        }
    
    }
}

File: dart/lib/api/miner_api.dart
part of swagger.api;



class MinerApi {
  final ApiClient apiClient;

  MinerApi([ApiClient apiClient]) : apiClient = apiClient ?? defaultApiClient;

  /// Get Claim Miner Message
  ///
  /// This endpoint lets a user get the message in order to claim a miner
  Future<GithubComApplicationResearchEstuaryApiV1ClaimMsgResponse> minerClaimMinerGet(String miner) async {
    Object postBody = null;

    // verify required params are set
    if(miner == null) {
     throw new ApiException(400, "Missing required param: miner");
    }

    // create path and map variables
    String path = "/miner/claim/{miner}".replaceAll("{format}","json").replaceAll("{" + "miner" + "}", miner.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    
    List<String> contentTypes = [];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = ["bearerAuth"];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return
          apiClient.deserialize(response.body, 'GithubComApplicationResearchEstuaryApiV1ClaimMsgResponse') as GithubComApplicationResearchEstuaryApiV1ClaimMsgResponse ;
    } else {
      return null;
    }
  }
  /// Claim Miner
  ///
  /// This endpoint lets a user claim a miner
  Future<GithubComApplicationResearchEstuaryApiV1ClaimResponse> minerClaimPost(MinerClaimMinerBody body) async {
    Object postBody = body;

    // verify required params are set
    if(body == null) {
     throw new ApiException(400, "Missing required param: body");
    }

    // create path and map variables
    String path = "/miner/claim".replaceAll("{format}","json");

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    
    List<String> contentTypes = ["*/*"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = ["bearerAuth"];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'POST',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return
          apiClient.deserialize(response.body, 'GithubComApplicationResearchEstuaryApiV1ClaimResponse') as GithubComApplicationResearchEstuaryApiV1ClaimResponse ;
    } else {
      return null;
    }
  }
  /// Set Miner Info
  ///
  /// This endpoint lets a user set miner info.
  Future<Map<String, String>> minerSetInfoMinerPut(MinerMinerSetInfoParams body, String miner) async {
    Object postBody = body;

    // verify required params are set
    if(body == null) {
     throw new ApiException(400, "Missing required param: body");
    }
    if(miner == null) {
     throw new ApiException(400, "Missing required param: miner");
    }

    // create path and map variables
    String path = "/miner/set-info/{miner}".replaceAll("{format}","json").replaceAll("{" + "miner" + "}", miner.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    
    List<String> contentTypes = ["*/*"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = ["bearerAuth"];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'PUT',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return
          new Map<String, String>.from(apiClient.deserialize(response.body, 'Map<String, String>')) ;
    } else {
      return null;
    }
  }
  /// Suspend Miner
  ///
  /// This endpoint lets a user suspend a miner.
  Future<Map<String, String>> minerSuspendMinerPost(MinerSuspendMinerBody body, String miner) async {
    Object postBody = body;

    // verify required params are set
    if(body == null) {
     throw new ApiException(400, "Missing required param: body");
    }
    if(miner == null) {
     throw new ApiException(400, "Missing required param: miner");
    }

    // create path and map variables
    String path = "/miner/suspend/{miner}".replaceAll("{format}","json").replaceAll("{" + "miner" + "}", miner.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    
    List<String> contentTypes = ["*/*"];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = ["bearerAuth"];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'POST',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return
          new Map<String, String>.from(apiClient.deserialize(response.body, 'Map<String, String>')) ;
    } else {
      return null;
    }
  }
  /// Unuspend Miner
  ///
  /// This endpoint lets a user unsuspend a miner.
  Future<Map<String, String>> minerUnsuspendMinerPut(String miner) async {
    Object postBody = null;

    // verify required params are set
    if(miner == null) {
     throw new ApiException(400, "Missing required param: miner");
    }

    // create path and map variables
    String path = "/miner/unsuspend/{miner}".replaceAll("{format}","json").replaceAll("{" + "miner" + "}", miner.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    
    List<String> contentTypes = [];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = ["bearerAuth"];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'PUT',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return
          new Map<String, String>.from(apiClient.deserialize(response.body, 'Map<String, String>')) ;
    } else {
      return null;
    }
  }
  /// Get all miners deals
  ///
  /// This endpoint returns all miners deals
  Future<String> publicMinersDealsMinerGet(String miner, { String ignoreFailed }) async {
    Object postBody = null;

    // verify required params are set
    if(miner == null) {
     throw new ApiException(400, "Missing required param: miner");
    }

    // create path and map variables
    String path = "/public/miners/deals/{miner}".replaceAll("{format}","json").replaceAll("{" + "miner" + "}", miner.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    if(ignoreFailed != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "ignore-failed", ignoreFailed));
    }
    
    List<String> contentTypes = [];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = ["bearerAuth"];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return
          apiClient.deserialize(response.body, 'String') as String ;
    } else {
      return null;
    }
  }
  /// Get miner stats
  ///
  /// This endpoint returns miner stats
  Future<String> publicMinersStatsMinerGet(String miner) async {
    Object postBody = null;

    // verify required params are set
    if(miner == null) {
     throw new ApiException(400, "Missing required param: miner");
    }

    // create path and map variables
    String path = "/public/miners/stats/{miner}".replaceAll("{format}","json").replaceAll("{" + "miner" + "}", miner.toString());

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    
    List<String> contentTypes = [];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = ["bearerAuth"];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return
          apiClient.deserialize(response.body, 'String') as String ;
    } else {
      return null;
    }
  }
}
